### PR TITLE
ACCUMULO-4546 Create default log message for IllegalTableTransitionException

### PR DESCRIPTION
**Pull request by jmark99**
ACCUMULO-4546 Create default log message for IllegalTableTransitionException

Updated the IllegalTableTransitionException class within TableManager.java to print
a default log message when exception is thrown. User also has option of supplying a
custom message. Created unit test.

Moved import in AccumuloTest.java to make mvn happy.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      throw new TableManager.IllegalTableTransitionException(oldState, newState);
+    } catch (IllegalTableTransitionException e) {
+      assertEquals(defaultMsg, e.getMessage());
+    }
+  }
+
+  @Test
+  public void testIllegalTableTransitionExceptionEmptyMessage() {
+    try {
+      throw new TableManager.IllegalTableTransitionException(oldState, newState, "");
```
Comment (keith-turner) [inline server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java]: In addition to `""`, could also try null.

File: server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java [hunk 2]
```diff
@@ -115,10 +115,21 @@ public TableState getTableState(String tableId) {
...
 
     public IllegalTableTransitionException(TableState oldState, TableState newState) {
+      this(oldState, newState, "");
+    }
+
+    public IllegalTableTransitionException(TableState oldState, TableState newState, String message) {
       this.oldState = oldState;
       this.newState = newState;
+
+      String defaultMessage = "Error transitioning from " + oldState + " state to " + newState + " state";
```
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java]: Could this be moved inside the else block?

File: server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java (hunk 1 above)
Comment (jmark99) [reply server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java]: I went back and forth between null and the empty string.  Both will work. I can go with either if you have a preference.

File: server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java (hunk 2 above)
Comment (jmark99) [reply server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java]: Yes...done.

File: server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java [hunk 3]
```diff
@@ -115,10 +115,22 @@ public TableState getTableState(String tableId) {
...
 
     public IllegalTableTransitionException(TableState oldState, TableState newState) {
+      this(oldState, newState, "");
+    }
+
+    public IllegalTableTransitionException(TableState oldState, TableState newState, String message) {
       this.oldState = oldState;
       this.newState = newState;
+
+      if (message != null && !message.isEmpty())
```
Comment (milleruntime) [inline server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java]: Could use StringUtils.isNotEmpty() from apache commons.  Or even static import it so it just reads if(isNotEmpty(message))

File: server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java (hunk 1 above)
Comment (keith-turner) [reply server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java]: I saw both were supported in the code.  I was thinking could have test for both.

Comment (jmark99) [reply server/base/src/test/java/org/apache/accumulo/server/tables/IllegalTableTransitionExceptionTest.java]: Added additional unit test.

File: server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java (hunk 3 above)
Comment (jmark99) [reply server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java]: Thanks, @milleruntime . I like that better.

**Pull request merged**